controllers/tektontasks_controller.go

```diff
@@ -129,6 +129,15 @@ func (r *tektonTasksReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, nil
 	}
 
+	multipleCrs, tektonTaskList, err := r.checkMultipleCRs(ctx, tektonRequest)
```

Comment:
You reduced checkMultipleCRs so much that you could basically just inline it here. Call `getCRList` here and replace `multipleCrs` in L137 with `len(crList) > 1`.